build/vega-lite-schema.json

```diff
@@ -2984,7 +2689,7 @@
               "$ref": "#/definitions/NumericFieldDefWithCondition"
             },
             {
-              "$ref": "#/definitions/NumericValueDefWithCondition"
+              "$ref": "#/definitions/NumericValueWithCondition"
```

Comment:
FYI, this is the only con I have about this PR, we need to rename `*ValueOrSignal` to something else.

However, we can't use ValueDef because 
given the hiding, `NumericValueOrSignalWithCondition` just refers to 
`NumericValueDefWithCondition` that already exists. We can't use the same name twice. Doing so will make JSON schema invalid.